pelita/maze_generator.py

```diff
@@ -252,53 +289,80 @@ def generate_half_maze(width, height, ngaps_center, bots_pos, rng=None):
     # use binary space partitioning
     rng = default_rng(rng)
 
-    # outer walls are top, bottom, left and right edge
-    walls = {(x, 0) for x in range(width)} | \
-            {(x, height-1) for x in range(width)} | \
-            {(0, y) for y in range(height)} | \
-            {(width-1, y) for y in range(height)}
+    # outer walls except the border
+    walls = (
+        # top
+        {(x, 0) for x in range(width // 2)}
+        # bottom
+        | {(x, height - 1) for x in range(width // 2)}
+        # left
+        | {(0, y) for y in range(height)}
+    )
 
-    # Generate a wall with gaps at the border between the two homezones
+    #
+    # BORDER SAMPLING
+    #
+    # generate a wall with gaps at the border between the two homezones
     # in the left side of the maze
 
-    # TODO: when we decide to break backward compatibility with the numpy version
-    # of create maze, this part can be delegated directly to generate_walls and
-    # then we need to rewrite mirror to mirror a set of coordinates around the center
-    # by discarding the lower part of the border
+    # start with a full wall at the left side of the border
+    pos = width // 2 - 1
```

Comment:
I find this name confusing. This seems to be something specific but calling it `pos` makes me think that it gets updated at a later point (also it is not a pos which should be 2d).